File: webapp/src/services/schemas.ts
export interface Token {
  access_token: string;
  token_type: string;
}

export interface TokenData {
  username?: string;
}

export interface UserBase {
  username: string;
  email: string;
}

export interface UserCreate extends UserBase {
  password: string;
}

export interface User extends UserBase {
  id: number;
  role_id?: number;
}

export interface QueryRequest {
  text: string;
}

export interface QueryResponse {
  responses: string[];
  status?: string;
}

File: _legacy_v1/core/schemas.py
from pydantic import BaseModel
from typing import List, Optional

# --- Esquemas para Tokens (JWT) ---

class Token(BaseModel):
    access_token: str
    token_type: str

class TokenData(BaseModel):
    username: Optional[str] = None

# --- Esquemas para Usuarios ---

class UserBase(BaseModel):
    username: str
    email: str

class UserCreate(UserBase):
    password: str

class User(UserBase):
    id: int
    role_id: Optional[int] = None

    class Config:
        from_attributes = True # Nuevo en Pydantic v2, reemplaza orm_mode

# --- Esquemas para la API Principal ---

class QueryRequest(BaseModel):
    text: str

class QueryResponse(BaseModel):
    responses: List[str]
    status: Optional[str] = None
